llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -587,6 +587,9 @@ void RequirementHandler::initAvailableCapabilitiesForOpenCL(
 void RequirementHandler::initAvailableCapabilitiesForVulkan(
     const SPIRVSubtarget &ST) {
   addAvailableCaps({Capability::Shader, Capability::Linkage});
+
+  // Provided by Vulkan version 1.0.
+  addAvailableCaps({Capability::Int16, Capability::Int64, Capability::Float64});
```

Comment:
Could we also add `Int8` and `Float16` so we have all the basic types in place?

Reply:
`Float16`: I'm going to handle this one separately in a follow up since it's not available in all Vulkan versions without an extension, and the HLSL translation is a bit more nuanced (actually translates to a `Float32` on the backend without the appropriate flag enabled) so I want to make sure our tests cover that case correctly.

`Int8`: I'm actually not sure what we use `Int8` for in DXC if anything because HLSL doesn't currently support 8-bit types. It probably wouldn't be any harm to add but I'd rather do it separately when meaningful tests can be added at the same time.

Reply:
Created an issue for tracking `Float16`: https://github.com/llvm/llvm-project/issues/66398

Reply:
Ok, fine for me.